pkgs/games/katago/default.nix

```diff
@@ -0,0 +1,100 @@
+{ stdenv
+, gcc8Stdenv
+, lib
+, libzip
+, boost
+, cmake
+, makeWrapper
+, fetchFromGitHub
+, linuxPackages ? null
+, cudnn ? null
+, cudatoolkit ? null
+, libGL_driver ? null
+, opencl-headers ? null
+, ocl-icd ? null
+, gperftools ? null
+, cudaSupport ? false
+, useTcmalloc ? false}:
+
+assert !cudaSupport || (
+  linuxPackages != null && 
+  libGL_driver != null && 
+  cudatoolkit != null &&
+  cudnn != null);
+
+assert cudaSupport || (
+  opencl-headers != null &&
+  ocl-icd != null);
+
+assert !useTcmalloc ||
+  gperftools != null;
+
+let
+  env = if cudaSupport 
+    then gcc8Stdenv
+    else stdenv;
+
+in env.mkDerivation rec {
+  pname = "katago";
+  version = "1.3.3";
+
+  src = fetchFromGitHub {
+    owner = "lightvector";
+    repo = "katago";
+    rev = "v${version}";
+    sha256 = "031pq0lhv2q46iip1d6nrhx5r669l3jc7gzdpw74j9xbhjvza9kz";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+  ];
+
+  buildInputs = [
+    libzip
+    boost
+  ] ++ lib.optionals cudaSupport [
+    linuxPackages.nvidia_x11
+    cudnn
+    libGL_driver
+  ] ++ lib.optionals (!cudaSupport) [
+    opencl-headers
+    ocl-icd
+  ] ++ lib.optionals useTcmalloc [
+    gperftools
+  ];
+
+  cmakeFlags = [
+    "-DNO_GIT_REVISION=ON"
+  ] ++ lib.optionals cudaSupport [
+    "-DUSE_BACKEND=CUDA"
+  ] ++ lib.optionals (!cudaSupport) [
+    "-DUSE_BACKEND=OPENCL"
+  ] ++ lib.optionals useTcmalloc [
+    "-DUSE_TCMALLOC=ON"
+  ];
+
+  preConfigure = ''
```

Comment:
probably the whole option could be `lib.optionalString`

Reply:
oh that's true, I don't know how I missed that on revision.

Reply:
Nope, you were correct, my comment is wrong.

Reply:
oof